### ISSUE TITLE
[verbose-decimal-constructor (FURB157)] doesn't trigger on `Decimal("1_000")`

### ISSUE THREAD

**Issue by monosans** [astral-sh/ruff#13807]
```python
Decimal(1000)  # OK
Decimal(1_000)  # OK
Decimal("1000")  # Triggers FURB157
Decimal("1_000")  # Should trigger FURB157 and become Decimal(1_000), but it doesn't
```

ruff v0.7.0

**Comment by AlexWaygood** [comment]
It seems like the parsing we do here is too naive: https://github.com/astral-sh/ruff/blob/4ecfe95295eb203ba771f344f52c8e61527a4c05/crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs#L77-L98

**Comment by AlexWaygood** [comment]
If there are more edge cases that we don't handle, it might be worth just using `ruff_python_parser::parse_expression` rather than reimplementing all the edge cases in the linter. If this is the only edge case we don't currently handle, however, it may be best to avoid doing that and just reimplement the logic in the linter -- using the parser might be quite expensive.

**Comment by dscorbett** [comment]
`Decimal` accepts more integer-valued strings than are valid as `int` literals. For example, `Decimal("1__2")` and `Decimal("_1_")` and `Decimal("١٢٣")` are valid. Should FURB157 trigger on those? If so, `ruff_python_parser::parse_expression` would not be enough.

**Comment by dylwil3** [comment]
Looks like this is the official regex: 

https://github.com/python/cpython/blob/322f14eeff9e3b5853eaac3233f7580ca0214cf8/Lib/_pydecimal.py#L6059-L6077

I can try to take this sometime this weekend (but if I don't, others should feel free to go for it!)

**Comment by m-aciek** [comment]
The fix in latest ruff (v0.13.1) converts `Decimal("1_000")` to `Decimal(1000)` instead of to `Decimal(1_000)`.

It becomes more problematic for bigger values, where thousand formatting really helps to read a number, like `Decimal(15_000_000)`.

Can the issue be reopened?
